pkg/util/compression/selector/zlib-and-zstd.go

```diff
@@ -0,0 +1,48 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build zlib && zstd
+
+// Package selector provides correct compression impl to fx
+package selector
+
+import (
+	common "github.com/DataDog/datadog-agent/pkg/util/compression"
+	implgzip "github.com/DataDog/datadog-agent/pkg/util/compression/impl-gzip"
+	implnoop "github.com/DataDog/datadog-agent/pkg/util/compression/impl-noop"
+	implzlib "github.com/DataDog/datadog-agent/pkg/util/compression/impl-zlib"
+	implzstd "github.com/DataDog/datadog-agent/pkg/util/compression/impl-zstd"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// NewCompressor returns a new Compressor based on serializer_compressor_kind
+// This function is called only when the zlib build tag is included
+func NewCompressor(kind string, level int) common.Compressor {
+	switch kind {
+	case common.ZlibKind:
+		return implzlib.New()
```

Comment:
Did you consider having two versions of `implzlib.New()`, one with  `//go:build zlib`  and another one with `//go:build !zlib`. It would allow to have  a single implementation of `NewCompressor`  (without build flag).

Reply:
Yes. I think this will come in a future PR. OTel are currently deciding on whether to use the native Go `zstd` implementation. Once this is determined we will decide how to structure this.

Reply:
OK. The idea here is to not have all the combinaisons of compression algorithms but of course can be done later.